### PR TITLE
Add if-can-be-merged rule

### LINKED ISSUE #441: [Rule] Adjacent identical IF expressions
Detect if there are IF blocks that could be merged:
```
IF  ${value} == 'True'
    Keyword
END
IF ${value} == 'True'
    Keyword2
END
```

### PR DESCRIPTION
**Pull request by bhirsz**
Closes #441

It's missing code for ignoring if with errors In else if branches (ie missing condition) but it's simple to implement - I will update PR later.

### CODE REVIEW

File: tests/atest/rules/misc/if-can-be-merged/test.robot [hunk 1]
```diff
@@ -0,0 +1,164 @@
+*** Test Cases ***
```
Comment (bhirsz) [inline tests/atest/rules/misc/if-can-be-merged/test.robot:1]: I see now I can also add tests with three identical ifs in a row.

File: tests/atest/rules/misc/if-can-be-merged/test.robot [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+    IF  ${condition} == 'True'
+        Step 1
+    ELSE IF  ${stuff}
+        Step 2
+    END
+    IF  ${condition} == 'True'
+        Step 2
+    END
+
+ElSE IF not adjacent
```
Comment (bhirsz) [inline tests/atest/rules/misc/if-can-be-merged/test.robot:114]: Duplicate, to be removed

File: tests/atest/rules/misc/if-can-be-merged/invalid.robot [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    END
+    IF
+        Keyword
+    END
+
+Two conditions
+    IF    ${condition}    Keyword
+        Keyword
+    END
+    IF    ${condition}    Keyword
```
Comment (mnojek) [inline tests/atest/rules/misc/if-can-be-merged/invalid.robot:14]: Why this is not caught?

File: tests/atest/rules/misc/if-can-be-merged/test.robot [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+        END
+    END
+
+
+
+    IF  ${condition} == 'True'
+        Step 2
+    END
+
+Singe nested If
```
Comment (mnojek) [inline tests/atest/rules/misc/if-can-be-merged/test.robot:54]:
```suggestion
Single nested If
```

File: tests/atest/rules/misc/if-can-be-merged/invalid.robot (hunk 3 above)
Comment (bhirsz) [reply tests/atest/rules/misc/if-can-be-merged/invalid.robot:14]: It's invalid.robot file - containing invalid syntax. Currently (for 4.0) you cannot use inline syntax for IFs and when you do
```
IF    ${condition}    Keyword
```
RF thinks yp're trying to use two conditions (which is not allowed).

**Review by mnojek — APPROVED** [review]

**Review by mnojek — APPROVED** [review]

**Pull request merged**